workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/TextControllerAsMutableState.kt

```diff
@@ -0,0 +1,29 @@
+@file:OptIn(WorkflowUiExperimentalApi::class)
+
+package com.squareup.workflow1.ui.compose
+
+import androidx.compose.runtime.*
+import com.squareup.workflow1.ui.TextController
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@Composable public fun TextController.asMutableState(): MutableState<String> {
+  return remember { mutableStateOf("") }.also { state ->
+    // Restart the effect if a different controller is passed in.
+    LaunchedEffect(this) {
+      state.value = textValue
```

Comment:
This won't run until after the first composition for the given text controller, which means you'll always get a frame with blank text. You need to initialize the state in the composition, ie when you create the mutable state.  I did it that way in the snippet for a reason. You should probably also pass the text controller as a key to remember so it's immediately updated if the controller changes as well.

Reply:
I refuse to admit how hard I looked for a version of remember that took a key without finding it four lines down. Remember kids, it's important to get a full night's sleep.